h/static/scripts/directive/annotation-share-dialog.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+var angular = require('angular');
+
+module.exports = function () {
+  return {
+    bindToController: true,
+    controllerAs: 'vm',
+    restrict: 'E',
+    template: require('../../../templates/client/annotation_share_dialog.html'),
+    controller: ['$timeout', '$scope', function ($timeout, $scope) {
+      this.copyToClipboard = function (event) {
+        var $container = angular.element(event.currentTarget).parent();
+        var shareLinkInput = $container.find('input')[0];
+
+        $scope.showCopyToClipboardMessage = true;
```

Comment:
Rather than maintain two separate pieces of state here (the message and whether to show a message), you could just make `copyToClipboardMessage` null when there is no message.